addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCCommonHandler.java

```diff
@@ -0,0 +1,279 @@
+     */
+    public abstract void setData(final byte[] data);
+
+    /**
+     * Checks if block name is valid.
+     *
+     * @param name Name of the LOGO! block to check
+     * @return True, if the name is valid and false otherwise
+     */
+    protected abstract boolean isValid(final String name);
+
+    /**
+     * Returns configured block kind.
+     *
+     * @return Configured block kind
+     */
+    protected abstract String getBlockKind();
+
+    /**
+     * Return number of channels dependent on configured Logo! family.
+     *
+     * @return Number of channels
+     */
+    protected abstract int getNumberOfChannels();
+
+    /**
+     * Calculate address for the block with given name.
+     *
+     * @param name Name of the LOGO! block
+     * @return Calculated address
+     */
+    protected int getAddress(final String name) {
+        int address = INVALID;
+
+        logger.debug("Get address of {} LOGO! for block {} .", getLogoFamily(), name);
+
+        final int base = getBase(name);
+        if (isValid(name) && (base != INVALID)) {
+            final String block = name.split("\\.")[0];
+            if (Character.isDigit(block.charAt(1))) {
+                address = Integer.parseInt(block.substring(1));
+            } else if (Character.isDigit(block.charAt(2))) {
+                address = Integer.parseInt(block.substring(2));
+            } else if (Character.isDigit(block.charAt(3))) {
+                address = Integer.parseInt(block.substring(3));
+            }
+        } else {
+            logger.warn("Wrong configurated LOGO! block {} found.", name);
+        }
+
+        return address;
+    }
+
+    /**
+     * Calculate address offset for given block name.
+     *
+     * @param name Name of the data block
+     * @return Calculated address offset
+     */
+    protected int getBase(final String name) {
+        int base = INVALID;
+
+        final String family = getLogoFamily();
+        logger.debug("Get base address of {} LOGO! for block {} .", family, name);
+
+        final String block = name.split("\\.")[0];
+        if (isValid(name) && !block.isEmpty()) {
+            final Map<?, Layout> memory = LOGO_MEMORY_BLOCK.get(family);
+            if (Character.isDigit(block.charAt(1))) {
+                base = memory.get(block.substring(0, 1)).address;
+            } else if (Character.isDigit(block.charAt(2))) {
+                base = memory.get(block.substring(0, 2)).address;
+            } else if (Character.isDigit(block.charAt(3))) {
+                base = memory.get(block.substring(0, 3)).address;
+            }
+        }
+
+        return base;
+    }
+
+    /**
+     * Checks if thing handler is valid and online.
+     *
+     * @return True, if handler is valid and false otherwise
+     */
+    protected boolean isThingOnline() {
+        final Bridge bridge = getBridge();
+        Objects.requireNonNull(bridge, "PLCCommonHandler: Bridge may not be null.");
```

Comment:
See above this does not really match with the declaration, maybe some null check would be more suitable